model-server/src/main/kotlin/org/modelix/model/server/handlers/IdsApiImpl.kt

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2024.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.modelix.model.server.handlers
+
+import io.ktor.server.application.Application
+import io.ktor.server.application.ApplicationCall
+import io.ktor.server.application.call
+import io.ktor.server.plugins.origin
+import io.ktor.server.response.respondText
+import io.ktor.server.routing.route
+import io.ktor.server.routing.routing
+import io.ktor.util.pipeline.PipelineContext
+import org.modelix.api.v2.IdsApi
+import org.modelix.authorization.getUserName
+import org.modelix.model.server.store.IStoreClient
+import org.modelix.model.server.store.LocalModelClient
+
+/**
+ * Implementation of the REST API that is responsible for handling client and server IDs.
+ */
+class IdsApiImpl(
+    private val repositoriesManager: IRepositoriesManager,
+    private val modelClient: LocalModelClient,
+) : IdsApi() {
+
+    constructor(modelClient: LocalModelClient) : this(RepositoriesManager(modelClient), modelClient)
+    constructor(storeClient: IStoreClient) : this(LocalModelClient(storeClient))
+    constructor(repositoriesManager: RepositoriesManager) : this(repositoriesManager, repositoriesManager.client)
+
+    private val storeClient: IStoreClient get() = modelClient.store
```

Comment:
This seems to be the only usage of `modelClient`. Does it make sense to replace `modelClient` with `storeClient` in the constructor?

Reply:
This pattern is currently copied from ModelReplicationServer, where this code comes from. In tests, the various forms of the constructors are used. One intermediate version is used to avoid instantiating the LocalModelClient twice and I am not aware of a different syntax to avoid this construct.

Generally, I'd like to get rid of most of this by replacing the hand-crafted test registration with a real ktor module. But I'd add that in a separate PR.

Reply:
This also addresses the previous comment for the constructors.